tests/integrationv2/test_compatibility_with_oqs_openssl.py

```diff
@@ -0,0 +1,272 @@
+        "expected_cipher": "ECDHE-RSA-AES256-GCM-SHA384",
+        "expected_kem": "NONE",
+    },
+    {
+        "server_ciphers": Ciphers.PQ_SIKE_TEST_TLS_1_0_2020_02,
+        "client_ciphers": Ciphers.ECDHE_RSA_AES256_GCM_SHA384,
+        "expected_cipher": "ECDHE-RSA-AES256-GCM-SHA384",
+        "expected_kem": "NONE",
+    },
+    {
+        "server_ciphers": Ciphers.PQ_SIKE_TEST_TLS_1_0_2020_02,
+        "client_ciphers": Ciphers.ECDHE_RSA_AES256_GCM_SHA384,
+        "expected_cipher": "ECDHE-RSA-AES256-GCM-SHA384",
+        "expected_kem": "NONE",
+    },
+    # The last set of vectors verify that a standard handshake can be completed when only one side support PQ
+    # by specifing a "mismatch" between PQ cipher preferences
+    {
+        "server_ciphers": Ciphers.KMS_PQ_TLS_1_0_2019_06,
+        "client_ciphers": Ciphers.ECDHE_RSA_AES256_GCM_SHA384,
+        "expected_cipher": "ECDHE-RSA-AES256-GCM-SHA384",
+        "expected_kem": "NONE",
+    },
+    {
+        "server_ciphers": Ciphers.KMS_PQ_TLS_1_0_2020_02,
+        "client_ciphers": Ciphers.ECDHE_RSA_AES256_GCM_SHA384,
+        "expected_cipher": "ECDHE-RSA-AES256-GCM-SHA384",
+        "expected_kem": "NONE",
+    },
+    {
+        "server_ciphers": Ciphers.KMS_TLS_1_0_2018_10,
+        "client_ciphers": Ciphers.ECDHE_RSA_AES256_GCM_SHA384,
+        "expected_cipher": "ECDHE-RSA-AES256-GCM-SHA384",
+        "expected_kem": "NONE",
+    },
+    {
+        "server_ciphers": Ciphers.KMS_TLS_1_0_2018_10,
+        "client_ciphers": Ciphers.ECDHE_RSA_AES256_GCM_SHA384,
+        "expected_cipher": "ECDHE-RSA-AES256-GCM-SHA384",
+        "expected_kem": "NONE",
+    },
+]
+
+def get_oqs_openssl_override_env_vars():
+    oqs_openssl_install_dir = os.environ["OQS_OPENSSL_1_1_1_INSTALL_DIR"]
+    
+    override_env_vars = dict()
+    override_env_vars["PATH"] = oqs_openssl_install_dir + "/bin"
+    override_env_vars["LD_LIBRARY_PATH"] = oqs_openssl_install_dir + "/lib"
+
+    return override_env_vars
+    
+    
+
+@pytest.mark.parametrize("vector", oqs_as_server_test_vectors)
+def test_oqs_openssl_as_server(managed_process, vector):
+    host = "localhost"
+    port = next(available_ports)
+
+    # We are manually passing the cipher flag to s2nc and s2nd.
+    # This is because PQ ciphers are specific to S2N at this point
+    # in time.
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host=host,
+        port=port,
+        insecure=True,
+        cipher=vector['client_ciphers'],
+        protocol=Protocols.TLS12)
+
+    server_options = ProviderOptions(
+        mode = Provider.ServerMode,
+        host=host,
+        port=port,
+        cipher=vector['server_ciphers'],
+        protocol=Protocols.TLS12,
+        cert=Certificates.RSA_4096_SHA512.cert,
+        key=Certificates.RSA_4096_SHA512.key,
+        env_overrides=get_oqs_openssl_override_env_vars())
+
+    server = managed_process(OpenSSL, server_options, timeout=5)
```

Comment:
Won't this use (regular) OpenSSL as a provider, and not OQS OpenSSL? Do you need to add something like `class OQS_OpenSSL(Provider)` to `providers.pq`?

Reply:
The OpenSSL Provider's main purpose is for converting a `ProviderOptions` Python structure into a String array that is runnable on the command line. That `ProviderOptions` to String array conversion is identical for both OpenSSL and OQS_OpenSSL since their `s_client` and `s_server` CLI interfaces are identical. The only difference between their `s_client` and `s_server` executables is their location on disk, and that some extra ciphers are allowed on OQS_OpenSSL.

https://github.com/awslabs/s2n/blob/master/tests/integrationv2/providers.py#L372-L416